src/navigation/courier/TaskListPage.js

```diff
@@ -29,108 +27,67 @@ const styles = StyleSheet.create({
   },
 });
 
-class TaskListPage extends Component {
-  completeSelectedTasks(selectedTasks) {
-    if (selectedTasks.length > 1) {
-      navigateToCompleteTask(
-        this.props.navigation,
-        this.props.route,
-        null,
-        selectedTasks,
-        true,
-      );
-    } else if (selectedTasks.length === 1) {
-      navigateToCompleteTask(
-        this.props.navigation,
-        this.props.route,
-        selectedTasks[0],
-        [],
-        true,
-      );
-    }
-  }
+const allowToSelect = task => {
+  return task.status !== 'DONE';
+};
 
-  allowToSelect(task) {
-    return task.status !== 'DONE';
+export default function TaskListPage({ navigation, route }) {
+  const selectedDate = useSelector(selectTaskSelectedDate);
+  const tasks = useSelector(selectFilteredTasks);
+  const tasksWithColor = useSelector(selectTasksWithColor);
+
+  const containerStyle = [styles.container];
+  if (tasks.length === 0) {
+    containerStyle.push(styles.containerEmpty);
   }
 
-  render() {
-    const { tasks, tasksWithColor, selectedDate } = this.props;
+  const { isFetching, refetch } = useGetMyTasksQuery(selectedDate, {
+    refetchOnFocus: true,
+  });
```

Comment:
As I understand, all screens with `refetchOnfocus` == true will try to fetch when they get focused, so we should be careful with using this option, otherwise we end up with a lot of requests. If  `refetchOnfocus` == false then there will be only one request and all other screens should get the data from the cache: https://redux-toolkit.js.org/rtk-query/usage/cache-behavior#default-cache-behavior 

I think, ideally, it would be cool to set a lifetime on each item in the cache, like 60 seconds, in that case we don't need `refetchOnfocus` we will just drop the old cached data and trigger a new request. I don't see any build-in ways to do it, so it seems we need to modify https://redux-toolkit.js.org/rtk-query/api/createApi#onquerystarted method ourselves

Reply:
There is `keepUnusedDataFor` https://redux-toolkit.js.org/rtk-query/usage/cache-behavior#reducing-subscription-time-with-keepunuseddatafor option, but I guess a component does not unsubscribe when it loses focus. If it's the case, then it will not work for us